forte/data/extractor/char_extractor.py

```diff
@@ -27,8 +32,28 @@
 
 class CharExtractor(BaseExtractor):
     r"""CharExtractor extracts feature from the text of entry.
-    Text will be splitted into characters.
+    Text will be split into characters.
+
+    Args:
+        config: An instance of `Dict` or
+            :class:`forte.common.configuration.Config` that provides all
+            configurable options. See :meth:`default_configs` for available
+            options and default values.
     """
+    def __init__(self, config: Union[Dict, Config]):
+        config = Config(config, self.default_configs(),
+                             allow_new_hparam=True)
```

Comment:
Do we need "allow_new_hparam" to be true?

Reply:
I think we don't need it. I will try to remove it.

There is a related question. What type of function should we implement "default_configs()"? static method, class method or instance method?

Reply:
Interesting question. So there is no final answer to this. We definitely will not use instance method, because that will make it hard to call it outside.

Static method and class method are similar. Class method makes it possible to extend. However, you can still achieve similar behaviors with static methods, but calling them explicitly. So actually I don't know which one is better.

Reply:
I check some other default_configs in forte. They implement it as class method. So I did the same.

And I found the allow_new_hparam need to be true. Because the default config of CharExtractor is only "max_char_length", new hparam like vocab_method etc need to be added to the config and then pass into the BaseExtractor.

Reply:
I extended the default config like other implementation. And I add default config for both CharExtractor and AttributeExtractor.

Now that the __init__ function is just calling the super class's __init__ function. Maybe I can delete the __init__ function in the sub class.

Reply:
yes, if there are no new items from the init, you can remove it.